RestaurantReservationSystem/RestaurantReservation.Db/Repositories/CustomerRepository.cs

```diff
@@ -0,0 +1,49 @@
+﻿using Microsoft.EntityFrameworkCore;
+using RestaurantReservation.Db.Contexts;
+using RestaurantReservation.Db.Entities;
+using RestaurantReservation.Db.Interfaces.Repositories;
+
+namespace RestaurantReservation.Db.Repositories;
+
+public class CustomerRepository : ICustomerRepository
+{
+    private readonly RestaurantReservationDbContext _context;
+
+    public CustomerRepository(RestaurantReservationDbContext context)
+    {
+        _context = context;
+    }
+
+    public async Task<IEnumerable<Customer>> GetAllAsync()
+    {
+        return await _context.Customers.ToListAsync();
+    }
+
+    public async Task<Customer?> GetByIdAsync(int id)
+    {
+        return await _context.Customers.FindAsync(id);
+    }
+
+    public async Task AddAsync(Customer customer)
+    {
+        _context.Customers.Add(customer);
+        await _context.SaveChangesAsync();
+    }
+
+    public async Task UpdateAsync(Customer customer)
+    {
+        _context.Customers.Update(customer);
+        await _context.SaveChangesAsync();
+    }
+
+    public async Task DeleteAsync(int id)
+    {
+        var customer = await _context.Customers.FindAsync(id);
+        if (customer is null)
```

Comment:
We can do an early return to avoid extra nesting.
```suggestion
        if (customer is null)
        {
            return;
        }
        _context.Customers.Remove(customer);
        await _context.SaveChangesAsync();
```